ddclient

```diff
@@ -2307,6 +2306,25 @@ sub get_ip {
     return $ip;
 }
 
+######################################################################
+## is_ipv4() validates if string is valid IPv4 address and only
+## a valid IPv4 address (no preceding or trailing spaces/characters)
+######################################################################
+sub is_ipv4 {
+    my ($value) = @_;
+    return (length($value) != 0) && ($value eq ipv4_match($value));
+}
+
+######################################################################
+## ipv4_match() extracts the first valid IPv4 address from given string
+######################################################################
+sub ipv4_match {
+    my ($value) = @_;
+    my $ip = undef;
+    $ip = $1 if $value =~ /\b((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))\b/ai;
+    return $ip;
```

Comment:
Similarly this will log a warning if `$value` is `undef`. How about this:

```suggestion
    (shift // '') =~ /\b((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))\b/ai;
    return $1;
```